addon/waiter-manager.ts

```diff
@@ -1,7 +1,17 @@
+import { registerWaiter } from '@ember/test';
 import { WaiterName, IWaiter, IPendingWaiterState } from './types';
 
 const WAITERS: Map<WaiterName, IWaiter> = new Map<WaiterName, IWaiter>();
 
+/**
+ * Backwards compatibility with legacy waiters system.
+ *
+ * We want to always register a waiter using the legacy waiter system, as right
+ * now if consumers are not on the right version of @ember/test-helpers, using
+ * this addon will result in none of these waiters waiting.
+ */
+registerWaiter(hasPendingWaiters);
```

Comment:
I actually think we need to guard this a tad:

```suggestion
if (Ember.Test) {
  registerWaiter(hasPendingWaiters);
}
```

Specifically we should use `if (Ember.Test)` (likely with eslint ignore shenanigans) so that we do not error in production builds (where invoking `registerWaiter` unconditionally like this will actually error) and we would prefer to avoid using `DEBUG` because of `ember test -prod`.

Reply:
I created https://github.com/rwjblue/ember-test-waiters/issues/14 to track ensuring we fail PR's that would break production non-testing builds.

Reply:
Ah yes. Good call.